### PR TITLE
assign a pauli rep to hadamard

### PR DESCRIPTION
**Pull request by timmysilv**
EDIT: I chatted with Jay, and now I'm unsure if I want this after all. See the drawbacks section below.

**Context:**
I was working on supporting op-arithmetic observables in lightning (see PennyLaneAI/pennylane-lightning#424), and figured what the heck, 9.5 times out of 10, the observable will be a Pauli sentence of sorts. Perhaps a common case for valid observables that does _not_ have a Pauli representation (today) would be if it contains a Hadamard. One simple way to fix this would be to enable representing Hadamard in the Pauli basis!

To confirm observables offered in PennyLane:
```pycon
>>> [obs.__name__ for obs in qml.operation.Observable.__subclasses__()]
['Tensor', 'CVObservable', 'Hadamard', 'PauliX', 'PauliY', 'PauliZ', 'Hermitian', 'SparseHamiltonian', 'Projector', 'Hamiltonian', 'GellMann']
```

**Description of the Change:**
Sets `Hadamard(w)._pauli_rep` to be $1/\sqrt(2)$ * `X(w) + Z(w)`

**Benefits:**
Enables arithmetic ops containing Hadamards to also have a `_pauli_rep`.

**Possible Drawbacks:**
Might encourage using the Pauli representation in some places where using the typical representation is better? The pauli module's main purpose was to improve operator simplification. So, take a look at this:
```pycon
>>> with_rep, without_rep = qml.Hadamard(0), qml.Hadamard(0)
>>> without_rep._pauli_rep = None
>>> qml.sum(with_rep, with_rep).simplify(), qml.sum(without_rep, without_rep).simplify()
((1.414213562373095*(PauliX(wires=[0]))) + (1.414213562373095*(PauliZ(wires=[0]))),
 2*(Hadamard(wires=[0])))
>>> qml.prod(with_rep, with_rep).simplify(), qml.prod(without_rep, without_rep).simplify()
(0.9999999999999998*(Identity(wires=[0])), Identity(wires=[0]))
```
tl;dr this may not be the best idea, I'll think on it more.

### CODE REVIEW

**Review by mudit2812 — APPROVED** [review]
Thanks Matt!
